stake-pool/program/src/state.rs

```diff
@@ -113,16 +113,26 @@ impl StakePool {
         )
         .ok()
     }
-    /// calculate the fee in pool tokens that goes to the manager
+
+    /// Calculate the fee in pool tokens that goes to the manager
+    ///
+    /// This function assumes that `reward_lamports` has not already been added
+    /// to the stake pool's `total_stake_lamports`
     pub fn calc_fee_amount(&self, reward_lamports: u64) -> Option<u64> {
         if self.fee.denominator == 0 {
             return Some(0);
         }
-        let pool_amount = self.calc_pool_tokens_for_deposit(reward_lamports)?;
+        let total_stake_lamports = self.total_stake_lamports.checked_add(reward_lamports)? as u128;
```

Comment:
If we widen to `u128` anyway, we can widen `self.total_stake_lamports` and `reward_lamports` before the add, and then the add is guaranteed not to overflow.

Reply:
The total lamports in the entire Solana system fits in a u64, so they're guaranteed by the runtime to not overflow.  You're right though, since we're widening either way, we can do it beforehand.